main.tf

```diff
@@ -17,7 +18,8 @@ resource "azurerm_service_plan" "this" {
   per_site_scaling_enabled     = var.per_site_scaling_enabled
   tags                         = var.tags
   worker_count                 = var.zone_balancing_enabled ? length(data.azurerm_location.region.zone_mappings) : var.worker_count
-  zone_balancing_enabled       = var.zone_balancing_enabled
+  # worker_count = var.worker_count # Allows flexibility of worker count, regardless of zone balancing
```

Comment:
Remove this commented out line as well

Reply:
I meant to leave a comment/question regarding this in the PR.

**Example**: Customer wants `2` instances in each zone
- `zone_balancing_enabled` = `true`
- `3` availability zones supported in selected region
- `worker_count` = `6` instances

With the current logic in `line 20` for `worker_count`, it doesn't allow for more than the number of availability zones if `zone_balancing_enabled` is `true`. So in this case, if `zone_balancing_enabled` is `true`, and `worker_count` = `6`, and the number of zones supported is 3, the module will only deploy 3 instances across the 3 zones

I think if it is simplified to `line 21`, it will allow for scenarios where customers want more that 3 instances.

I know this case/functionality might not seem super important, but it's in one of the quickstarts and you may receive an issue later on. Let me know what you think?

Reply:
Ah yes I do see what you're saying. That's a very valid scenario. 

If we allow the customer to set the worker_count and zone redundancy is enabled, there's no validation for the multiplier, but if we add a new condition that checks if worker_count is a multiple of the number of available azs such as 
`worker_count = var.zone_balancing_enabled ? ceil(var.worker_count / length(data.azurerm_location.region.zone_mappings)) * length(data.azurerm_location.region.zone_mappings) : var.worker_count`

This would also force the worker_count to be a multiple of the azs and the customer wouldn't have to redeploy with a new worker_count. 

Thoughts?

Reply:
That works for me as well 👍 I can update this branch / pr.